lib/practice_exercises.rb

```diff
@@ -1,13 +1,28 @@
 
-# Time Complexity: ?
-# Space Complexity: ?
+# Time Complexity: O(n)
+# Space Complexity: O(1)
 def is_palindrome(string)
-  raise NotImplementedError, "Not implemented yet"
+  string.downcase! #cannot have this in front of gsub or it will replace the word with nil if there are already downcase letters
+  string.gsub!(/[^A-Za-z]/, '')
+
+  string.each_char.with_index do |letter, index|
+    if letter != string[string.length - 1 - index]
+      return false
+    end
+  end
+
+  return true
 end
 
-# Time Complexity: ?
-# Space Complexity: ?
+# Time Complexity: O(n*m)
+# Space Complexity: O(1)
 def longest_prefix(strings)
```

Comment:
👍 Since you are building a prefix which, worst-case, is the length of the shortest string, the space complexity is O(m).